wdl-engine/src/eval/v1/task.rs

```diff
@@ -0,0 +1,929 @@
+                                document,
+                                execution.as_ref(),
+                                &scopes,
+                                task,
+                                section,
+                                inputs,
+                            )?);
+                        }
+                    }
+
+                    current += 1;
+                }
+
+                let requirements = requirements.unwrap_or_default();
+                let hints = hints.unwrap_or_default();
+
+                // TODO: check call cache for a hit. if so, skip task execution and use cache
+                // paths for output evaluation
+
+                let status_code = execution.spawn(command, &requirements, &hints)?.await?;
+
+                // TODO: support retrying the task if it fails
+
+                let mut evaluated = EvaluatedTask::new(execution.as_ref(), status_code)?;
+
+                // Update the task variable's return code
+                if version >= SupportedVersion::V1(V1::Two) {
+                    let task = scopes[TASK_SCOPE_INDEX]
+                        .get_mut(TASK_VAR_NAME)
+                        .unwrap()
+                        .as_task_mut()
+                        .unwrap();
+                    task.set_return_code(evaluated.status_code);
+                }
+
+                // Use a closure that returns an evaluation result for evaluating the outputs
+                let mut outputs = || -> EvaluationResult<Outputs> {
+                    evaluated.handle_exit(&requirements)?;
+
+                    for index in &nodes[current..] {
+                        match &graph[*index] {
+                            TaskGraphNode::Output(decl) => {
+                                self.evaluate_output(
+                                    document,
+                                    &mut scopes,
+                                    task,
+                                    decl,
+                                    &evaluated,
+                                )?;
+                            }
+                            TaskGraphNode::Input(decl) => {
+                                self.evaluate_input(
+                                    document,
+                                    execution.as_ref(),
+                                    &mut scopes,
+                                    task,
+                                    decl,
+                                    inputs,
+                                )?;
+                            }
+                            TaskGraphNode::Decl(decl) => {
+                                self.evaluate_decl(
+                                    document,
+                                    execution.as_ref(),
+                                    &mut scopes,
+                                    task,
+                                    decl,
+                                )?;
+                            }
```

Comment:
How can an input/decl follow execution? Shouldn't these have been evaluated already?

Reply:
Not if they're either:

1. unused (i.e. have no outgoing dependency edges)
2. have only outgoing edges to outputs (i.e. the command and requirements/hints/runtime sections do not reference the declaration

In this case the input or declaration would be evaluated after the command in topological ordering.

Reply:
Do they even need to be evaluated if they aren't used? Is that just wasted computation?

Reply:
Technically, the unused ones don't, but those referenced only by the outputs definitely do need to be evaluated; I figure we should probably evaluate the unused ones anyway for errors; we already emit a warning informing users that they can be removed with the justification that it would, in theory, reduce evaluation time.

Reply:
That makes sense. I suppose a decent compromise is what you have implemented here. If one of these unused decls causes an error but the task otherwise completed successfully, the results could be salvaged from the run dir.

Q: What about "silently used" inputs like `.bam.bai` files? Those might not be referenced anywhere in the command/output, but certainly need to be localized for successful task completion.

Reply:
I think I'll need to revisit path mapping in the future as right now files need to be transitively referenced by the command for them to be mapped into a container environment; so while we have a heuristic to suppress an unused input declaration on an input named `foo_bai`, we won't actually mount the file into the container unless the command references it.

Note, though, that it'll work now if you use a struct value to group the related paths together, since we map all the struct members if the struct value is referenced by the command.

This will all need to be tweaked a bit once we're doing more than local execution and mounting of files is required.

Reply:
Also note that initial workflow/task inputs that reference URLs isn't yet implemented, either. I can open a feature issue for it.

Reply:
> I think I'll need to revisit path mapping in the future as right now files need to be transitively referenced by the command for them to be mapped into a container environment; so while we have a heuristic to suppress an unused input declaration on an input named `foo_bai`, we won't actually mount the file into the container unless the command references it.

Yea, I think we can move forward with this PR without that, but it will most certainly need to be addressed in a follow up. I think the answer here is to just always localize/mount `File` inputs? I suppose it could be a wasted computation, but it's what I would expect as a user.

> Note, though, that it'll work now if you use a struct value to group the related paths together, since we map all the struct members if the struct value is referenced by the command.

This is an interesting caveat. Thank you for sharing! With this snippet of info, I actually think this corner case will be quite rare in the wild. With the WDLs I've reviewed, I think it's most common to "package" a "main" file and their indices/accessory files in a Struct. `workflows` is in the minority in that we do _not_ use structs this way. We have separate inputs for "main" files and their accessory files, _but_ we always localize them with `ln -s` in the command block as a sanity check. That should be unnecessary, but we've run into issues with some runners not handling file localization properly.

> This will all need to be tweaked a bit once we're doing more than local execution and mounting of files is required.

Agree we can come back 👍 
In the meantime, can you open an issue to track this? Don't want this lost in the shuffle.

Reply:
Opened https://github.com/stjude-rust-labs/wdl/issues/267.